File: modules/example/src/main/java/dev/httpmarco/polocloud/modules/example/ExampleModule.java
package dev.httpmarco.polocloud.modules.example;

import dev.httpmarco.polocloud.node.modules.CloudModule;
import lombok.extern.log4j.Log4j2;

@Log4j2
public class ExampleModule implements CloudModule {

    @Override
    public void onEnable() {
        log.info("Example Module enabled");
    }

    @Override
    public void onDisable() {
        log.info("Example Module disabled");
    }
}

File: modules/example/src/main/kotlin/dev/httpmarco/polocloud/modules/example/ExampleModule.kt
package dev.httpmarco.polocloud.modules.example

import dev.httpmarco.polocloud.shared.module.PolocloudModule
import dev.httpmarco.polocloud.shared.polocloudShared

class ExampleModule : PolocloudModule {
    override fun onEnable() {
        // Here you can add your module's initialization logic
        polocloudShared.logger().info("Example module enabled.")
    }

    override fun onDisable() {
        // Here you can add your module's cleanup logic
        polocloudShared.logger().info("Example module disabled.")
    }
}